module-system/module-implementations/sov-bank/src/call.rs

```diff
@@ -137,29 +137,41 @@ impl<C: sov_modules_api::Context> Bank<C> {
         Ok(CallResponse::default())
     }
 
-    /// Mints the `coins` set by the address `minter_address`. If the token address doesn't exist return an error.
+    /// Mints the `coins`to the address `mint_to_address` using `context.sender()` as the authorizer.
+    /// Returns an error if the token address doesn't exist or `context.sender()` is not authorized to mint tokens.
+    /// Calls the [`Token::mint`] function and update the `self.tokens` set to store the new balance.
+    pub fn mint_from_eoa(
+        &self,
+        coins: &Coins<C>,
+        mint_to_address: &C::Address,
+        context: &C,
+        working_set: &mut WorkingSet<C::Storage>,
+    ) -> Result<CallResponse> {
+        self.mint(coins, mint_to_address, context.sender(), working_set)
+    }
+
+    /// Mints the `coins` to the address `mint_to_address` if `authorizer` is an allowed minter.
+    /// Returns an error if the token address doesn't exist or `context.sender()` is not authorized to mint tokens.
     /// Calls the [`Token::mint`] function and update the `self.tokens` set to store the new minted address.
     pub fn mint(
         &self,
         coins: &Coins<C>,
-        minter_address: &C::Address,
-        context: &C,
+        mint_to_address: &C::Address,
+        authorizer: &C::Address,
         working_set: &mut WorkingSet<C::Storage>,
     ) -> Result<CallResponse> {
```

Comment:
Does it have to return `CallResponse`, now it is a public fn so I think `()` should be enough?

Reply:
I'll change to `()`.